travel-event-host-app/src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -4,18 +4,59 @@ import { connectMongoDB } from '@/lib/mongodb';
 import User from '@/schemas/user';
 import { compare } from 'bcrypt';
 
+type SessionUser = {
+  name?: string | null | undefined;
+  _id: string;
+  firstName?: string;
+  imageUrl?: string;
+  lastName?: string;
+  email: string;
+  location: {
+    country: string;
+    state: string;
+    city: string;
+    coords: {
+      lat: number;
+      long: number;
+    };
+  };
+  eventIds: string[];
+  admin?: boolean;
+};
+
 const handler = NextAuth({
   callbacks: {
     async signIn({ user, account }) {
       return true;
     },
     async jwt({ token, account, profile }) {
+      await connectMongoDB();
```

Comment:
In this jwt, can you write some comments to explain what is happening, so everyone can understand?

Reply:
sure